File: frontend/src/features/products/types/product.ts
export interface Category {
  id: string;
  name: string;
  description?: string;
  icon?: string;
  parentId?: string;
}

export type ProductCondition = 'NEW' | 'LIKE_NEW' | 'GOOD' | 'FAIR';
export type SellType = 'BUY_NOW' | 'AUCTION';
export type ProductStatus = 'ACTIVE' | 'SOLD' | 'CANCELLED' | 'HIDDEN' | 'DELETED';

export interface Product {
  id: string;
  sellerId: string;
  sellerName: string;
  sellerAvatar?: string;
  categoryId: string;
  categoryName: string;
  title: string;
  description: string;
  condition: ProductCondition;
  sellType: SellType;
  price: number;
  quantity: number;
  imageUrl?: string;
  videoUrl?: string;
  status: ProductStatus;
  location?: string;
  createdAt: string;
  auctionEndTime?: string;
  boostedUntil?: string;
  currentHighestBid?: number;
  bidCount?: number;
  isLive?: boolean;
}

export interface CreateProductRequest {
  title: string;
  description: string;
  price: number;
  categoryId: string;
  condition: ProductCondition;
  sellType: SellType;
  imageUrl?: string;
  videoUrl?: string;
  location?: string;
  auctionDurationDays?: number;
}

File: backend/src/main/java/com/ecommerce/thriftauction/features/product/entity/Product.java
package com.ecommerce.thriftauction.features.product.entity;

import com.ecommerce.thriftauction.features.auth.entity.User;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seller_id", nullable = false)
    private User seller;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Builder.Default
    private Integer quantity = 1;

    @Enumerated(EnumType.STRING)
    private ProductCondition condition;

    @Enumerated(EnumType.STRING)
    private SellType sellType;

    @Column(nullable = false)
    private BigDecimal price;

    @Enumerated(EnumType.STRING)
    private ProductStatus status;

    private String imageUrl;

    private String videoUrl;

    private String location;

    @Column(columnDefinition = "boolean default false")
    private Boolean isHidden = false;

    @Column(columnDefinition = "boolean default false")
    @Builder.Default
    private Boolean isLive = false;

    private LocalDateTime boostedUntil;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
